src/commands/createFunction/FunctionListStep.ts

```diff
@@ -136,7 +136,10 @@ export class FunctionListStep extends AzureWizardPromptStep<IFunctionWizardConte
                 data: <IFunctionTemplate | TemplatePromptResult><unknown>undefined,
                 onPicked: () => { /* do nothing */ }
             })
-        } else if (language === ProjectLanguage.CSharp || language === ProjectLanguage.Java || (language === ProjectLanguage.Python && !isPythonV2Plus(language, languageModel)) || language === ProjectLanguage.TypeScript) {
+        } else if (language === ProjectLanguage.CSharp ||
+            language === ProjectLanguage.Java ||
+            (language === ProjectLanguage.Python && !isPythonV2Plus(language, languageModel)) ||
+            (language === ProjectLanguage.TypeScript) && !isNodeV4Plus(context)) {
```

Comment:
Not sure it makes a difference, but I think you meant to do this:
```suggestion
            (language === ProjectLanguage.TypeScript && !isNodeV4Plus(context))) {
```